### PR TITLE
feat(clickhouse sink): add format

### PR DESCRIPTION
**Pull request by gabriel376**
This PR adds initial support to [formats](https://clickhouse.com/docs/en/interfaces/formats) in the [Clickhouse sink](https://vector.dev/docs/reference/configuration/sinks/clickhouse/).

More format options can be easily added in the future to the `clickhouse::config::Format` enum.

Usage example:
```toml
[sources.logs]
type = "demo_logs"
format = "json"

[sinks.clickhouse]
type = "clickhouse"
inputs = ["logs"]
endpoint = "http://localhost:8123"
table = "mytable"
format = "json_as_object"
```

Resolves https://github.com/vectordotdev/vector/issues/15813

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
Apologies, could you add a changelog entry for this change? You can see https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md for the process.

**Comment by gabriel376** [comment]
@jszwedko thank you for the quick response.
I added a changelog fragment in commit https://github.com/vectordotdev/vector/pull/19739/commits/9aa202cbc5268f0c7a45c044cb031db0a2f631be.
Let me know if anything else is missing or can be improved.

**Review by jszwedko — APPROVED** [review]
Thanks @gabriel376 !

**Comment by jszwedko** [comment]
@gabriel376 apologies, it looks like I missed some clippy errors in CI. Do you think you could take a look?

**Comment by gabriel376** [comment]
@jszwedko Just pushed a new commit to ignore some `clippy` warnings. Please let me know if you have suggestions for a better fix.

File: src/sinks/clickhouse/config.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ use crate::{
 #[derive(Clone, Copy, Debug, Derivative, Eq, PartialEq)]
 #[serde(rename_all = "snake_case")]
 #[derivative(Default)]
+#[allow(clippy::enum_variant_names)]
```
Comment (jszwedko) [inline src/sinks/clickhouse/config.rs:26]: 👍 Makes sense to keep the variant names the same as what Clickhouse calls them.

**Review by jszwedko — APPROVED** [review]
Thanks @gabriel376 !

**Pull request merged**